### DISCUSSION TITLE
Looping through and displaying other file's metadata

### DISCUSSION THREAD

**Discussion by SoumayaMauthoorMOJ** [squidfunk/mkdocs-material#5101] [Q&A] [answered]
Is it possible to reference the metadata in other files? 

I have added an `adr` keyword to the front-matter of my Architecture Decision Record markdown files:

```
---
adr: 
    number: "0001"
    status: Draft
---
```

I have updated the `content` block in main.html so that this information is displayed before the title:

```
{% block content %}
  {% if page.meta and page.meta.adr %}
    <table>
      <tr>
        <th>Number</th>
        <th>{{page.meta.adr.number}}</th>
      </tr>
      <tr>
        <th>Status</th>
        <th>{{page.meta.adr.status}}</th>
      </tr>
    </table>
  {% endif %}
  {{ super() }}
{% endblock %}
```

I was wondering whether it was possible to loop through these files and render them on a designated file e.g.:

```
---
adr_root: true
---
```

```  
{% block content %}
 {{ super() }}

{% if page.meta and page.meta.adr_root %}
    {%  for page in pages %}
      {% if page.meta and page.meta.adr %}
        * [{{page.title}}]
      {% endif %}
    {% endfor %}
{% endif %}

{% endblock %}
```

Or would I have to create my own plugin similar to the [tag](https://github.com/squidfunk/mkdocs-material/blob/3b921f76a3bb268f169dd800ba59bbad7dedeb83/material/plugins/tags/plugin.py) plugin

**Comment by squidfunk** [comment, marked as answer]
You should be able to access the metadata of all pages via the `nav` object, as this contains all pages,so something like this might work (but YMMV depending on what you want to achieve):

``` jinja2
{% block content %}
  {% for nav_item in nav %}
    {{ nav_item.meta.adr_root }}
  {% endfor %}
{% endblock %}
```

Also, you would need to recursively loop through children. See the [`nav_item`](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/nav-item.html) partial [for guidance](https://github.com/squidfunk/mkdocs-material/blob/59df2e433d4d3007b407afbb7857c3f51c5c6d32/src/partials/nav-item.html#L112-L117).

**Reply by SoumayaMauthoorMOJ** [reply]
Hello @squidfunk thanks for the suggestion but in the end I went for a completely different approach based on a judicious use of tags (which is a very cool plug-in by the way). You can see a [demo](https://ministryofjustice.github.io/mkdocs-tech-docs-template/ADRs/) if you're interested. It does mean I can't add other kinds of tags until I get my organisation to approve mkdocs-insiders so fingers crossed!

**Reply by squidfunk** [reply]
Wow, what a nice mod! Love it! I also love how you adapted the tag functionality. I'm planning to extend the plugin with additional functionality, any input is appreciated! See https://github.com/squidfunk/mkdocs-material/discussions/5094#discussioncomment-5113497

**Reply by SoumayaMauthoorMOJ** [reply]
I would need to first install mkdodcs-insiders and play around with the `tag_extra_files` functionality before I can comment on #5094. However it would be cool to have the ability to specify a background color for a specified tag, like you can specify an icon. I could raise as an "idea" if you like?

**Reply by squidfunk** [reply]
This should already be possible with custom CSS:

``` css
.md-tag--accepted { /* <- Tag identifier */
  background-color: red;
}
```

**Reply by squidfunk** [reply]
Small update: I just changed `md-tag-icon--${ident}` to `md-tag--${ident}` in 4d7291f25, because it's semantically better.

I adjusted the example above.

**Reply by SoumayaMauthoorMOJ** [reply]
Amazing thank you so much! I have separated the colors in a separate [tags-color.css](https://github.com/ministryofjustice/mkdocs-tech-docs-template/blob/main/tech_docs_template/assets/stylesheets/tags-color.css) to make it easier to differentiate from the rest of the extra stylesheet customisation :-)

**Reply by squidfunk** [reply]
9.1.0 is out, which includes the changes mentioned in https://github.com/squidfunk/mkdocs-material/discussions/5101#discussioncomment-5114543.

**Reply by SoumayaMauthoorMOJ** [reply]
Perfect thanks for letting me know will update :-)

**Reply by vlebert** [reply]
hi @squidfunk 
I do have the same need. What if my pages are not in the navigation ? Neither GPT or Claude or Codestral could help me :)

**Reply by kamilkrzyskow** [reply]
Have you tried using pages @vlebert?
- https://www.mkdocs.org/dev-guide/themes/#pages

Just be cautious to not override the `page`, as this is also a global context variable. Also better don't use `post`/`posts`, as blog posts from the plugin are set via those:
```jinja2
{% block content %}
  {% for loop_file in pages %}
    {{ loop_file.page.meta }}
  {% endfor %}
{% endblock %}
```

**Reply by vlebert** [reply]
Hi @kamilkrzyskow 

When using pages, i can iterate on pages but properties such as title or meta are empty

Or maybe i am missing something...

**Reply by kamilkrzyskow** [reply]
Oh sorry @vlebert, I didn't read the fine print, which was the whole description...
> A flat list of File objects for all pages in the project. This list can contain pages not included in the global [navigation](https://www.mkdocs.org/dev-guide/themes/#nav) and may not match the order of pages within that navigation. The [page](https://www.mkdocs.org/dev-guide/themes/#page) object for each File can be accessed from file.page.

So those are Files not Pages, so to access the meta, you need to first access the page, I've edited the example above to use the `file.page` structure. If that still doesn't work, please provide a minimal example zip archive I could run and modify ✌️

**Reply by vlebert** [reply]
Thanks @kamilkrzyskow 
I think my issue from the begining came from using a "page" variable name in my for loop :)
